tools/mercurial/creating_mercurial_patch.md

```diff
@@ -0,0 +1,295 @@
+
+```
+pick f6f70f6de69c 358597 Bug 123456 - [ur] Search engine setup for Firefox fo...
+roll 8088fd8658fd 358598 Fix searchplugin name
+
+...
+```
+
+Push again to MozReview
+
+```BASH
+$ hg push review
+```
+
+You can also use `hg histedit` to reword a commit message (set the commit line to `edit`). Just remember to complete the `histedit` after commit.
+
+```BASH
+$ hg commit -m "Some changes"
+$ hg histedit --continue
+```
+
+If necessary, you can rebase against another bookmark, like `central` or `inbound`
+
+```BASH
+$ hg rebase -d central
+```
+
+More information about this workflow are available in the following pages:
+* https://mozilla-version-control-tools.readthedocs.io/en/latest/hgmozilla/firefoxworkflow.html
+* https://www.mercurial-scm.org/wiki/Bookmarks
+
+### Landing the patch
+
+Once the patch has been reviewed, you have two options:
+* If you have L3 access to the repository, you can use Autoland to land your commit directly from MozReview. If your reviewer has it, you can ask them to land.
+* You can set the `checkin-needed` keyword in the bug, and sheriffs will land it from you.
+
+## Creating a patch using Queues
+
+### Creating a patch
+
+Consider the following example: setting up productization for Urdu (ur) on Firefox desktop (l10n repository). The first step is to create a `region.properties` file. Move into the repository folder and check its status:
+
+```BASH
+$ cd ~/mozilla/mercurial/l10n/ur/l10n-central
+
+$ hg status
+? browser/chrome/browser-region/region.properties
+```
+
+File is brand new and needs to be added:
+
+```BASH
+$ hg add browser
+
+$ hg status
+A browser/chrome/browser-region/region.properties
+```
+
+You need to assign a name to this patch, it’s easy to use a reference to the bug number: for example, if the bug number is 123456, the file could be called `bug123456.patch` (note the added extension `.patch`).
+
+```BASH
+$ hg qnew bug123456.patch
+```
+
+At this point you will be asked to provide a commit message for your patch (in `nano` if you followed the instructions to set up the environment): write your commit message, then press `CTRL+O` to save the file, `enter` to confirm the proposed filename, and finally `CTRL+X` to exit the editor.
+
+The commit message should be the same as the bug, for example `Bug 123456 - Set up searchplugins for "ur" and Firefox desktop`.
+
+You’ready to *pop* the patch out of the queue. Since it’s likely that there are no other patches, you can pop them all with `-a`.
+
+```BASH
+$ hg qpop -a
+popping bug123456.patch
+patch queue now empty
+```
+
+The patch is stored inside the `.hg/patches` folder in the root of the repository (in the suggested setup, the full path would be `~/mozilla/mercurial/l10n-central/.hg/patches`). You can copy the file through the command line or the file explorer. For example, on macOS you can open the folder in Finder by typing:
```

Comment:
Not in this case. The example for Queues uses the l10n repository `cd ~/mozilla/mercurial/l10n/ur/l10n-central`, so this needs be fixed anyway (it should be `~/mozilla/mercurial/l10n/ur/l10n-central/.hg/patches`, missing `ur`).

Do you think it would be clearer to use the same files of the MozReview section?